packages/fast-element/src/directives/binding.ts

```diff
@@ -0,0 +1,168 @@
+import { Directive } from "./directive";
+import { Expression } from "../interfaces";
+import { Subscriber } from "../observation/subscriber-collection";
+import {
+    Observable,
+    GetterInspector,
+    inspectAndEvaluate,
+} from "../observation/observable";
+import { DOM } from "../dom";
+import { Behavior } from "./behavior";
+
+export const enum BindingType {
+    attribute = 0,
+    booleanAttribute = 1,
+    text = 2,
+    property = 3,
+    trigger = 4,
+}
+
+function normalBind(this: BindingBehavior, source: unknown) {
+    this.source = source;
+    this.updateTarget(inspectAndEvaluate(this.expression, source, context, this));
+}
+
+function triggerBind(this: BindingBehavior, source: unknown) {
+    this.source = source;
+    this.target.addEventListener(this.targetName!, this, true);
+}
+
+function normalUnbind(this: BindingBehavior) {
+    if (this.record !== null) {
+        this.record.unsubscribe(this);
+    }
+
+    this.source = null;
+}
+
+function triggerUnbind(this: BindingBehavior) {
+    this.target.removeEventListener(this.targetName!, this, true);
+    this.source = null;
+}
+
+function updateAttributeTarget(this: BindingBehavior, value: unknown): void {
```

Comment:
Note to self: this implementation is incorrect. Need to fix this before merging.